### PR TITLE
 teuthology command: introduce --interactive-on-error flag

### PR DESCRIPTION
**Pull request by ideepika**

### CODE REVIEW

File: teuthology/run_tasks.py [hunk 1]
```diff
@@ -141,6 +141,7 @@ def run_tasks(tasks, ctx):
 
         if ctx.config.get('interactive-on-error'):
             ctx.config['interactive-on-error'] = False
+            del ctx.config['archive_path']
```
Comment (kshtsk) [inline teuthology/run_tasks.py:144]: Why this is required here?

**Comment by kshtsk** [comment]
Seems like it does not do anything yet.

File: teuthology/run_tasks.py (hunk 1 above)
Comment (ideepika) [reply teuthology/run_tasks.py]: I observed that archive_path used to hider with manual interactive on error method when using orig.config.yaml of a teuthology job, but you are right I can verify it's not required and call runs smooth without this configuration as well, removed

**Comment by ideepika** [comment]
> Seems like it does not do anything yet.

I ran master with : 
```
./virtualenv/bin/teuthology --lock --block test.yaml   
```
result:
```
duration: 917.2966458797455
failure_reason: expected str, bytes or os.PathLike object, not NoneType
flavor: basic
owner: ideepika@x1carbon
status: fail
success: false

```
exhibiting the general nature of `teuthology` command

using --interactive-on-error:
```
~/teuthology add-interactive-on-error *2 ?2 ❯ ./virtualenv/bin/teuthology -v --lock --block test.yaml --interactive-on-error 
```
https://pastebin.com/uidY2Sd4

File: scripts/run.py [hunk 2]
```diff
@@ -23,6 +23,10 @@
   --name NAME                    name for this teuthology run
   --suite-path SUITE_PATH        Location of ceph-qa-suite on disk. If not specified,
                                  it will be fetched
+  --interactive-on-error         drop to a python shell on failure, which will keep
+                                 job in halt state, developer can then ssh to deployment
```
Comment (kshtsk) [inline scripts/run.py:27]: deployment machine => target machines

File: docs/detailed_test_config.rst [hunk 3]
```diff
@@ -232,13 +232,16 @@ to explore the system as is. Adding a top-level::
...
-option, any *task* that fails, will have the ``interactive`` task
-called after it. This means that before any cleanup happens, you get a
-chance to inspect the system -- both through Teuthology and via extra
-SSH connections -- and the cleanup completes only when you choose so.
+as a config file for teuthology or passing ``--interactive-on-error`` flag will
+make that possible. With this option, any *task* that fails, will have the
+``interactive`` task called after it.  This means that before any cleanup
+happens, you get a chance to inspect the system -- both through Teuthology and
+via extra SSH connections -- and the cleanup completes only when you choose so.
 Just exit the interactive Python session to continue the cleanup.
```
Comment (kshtsk) [inline docs/detailed_test_config.rst:240]: I would not change this paragraph at all. Maybe just add a single sentence to the end:
The interactive-on-error can be enabled with `--interactive-on-error` option with ``teuthology`` command.

File: scripts/openstack.py [hunk 4]
```diff
@@ -224,13 +224,15 @@ def get_openstack_parser():
     )
     parser.add_argument(
         '--controller-ram',
-        help='override default minimum RAM (in megabytes) when selecting flavor for teuthology VM',
```
Comment (kshtsk) [inline scripts/openstack.py:227]: All changes in this file are not related to the PR.

Comment (kshtsk) [reply scripts/openstack.py:227]: The message in the commit does not correspond to changes. This does not look like cleanup.

File: docs/detailed_test_config.rst (hunk 3 above)
Comment (ideepika) [reply docs/detailed_test_config.rst:240]: updated

File: scripts/run.py (hunk 2 above)
Comment (ideepika) [reply scripts/run.py]: updated.

File: scripts/openstack.py (hunk 4 above)
Comment (ideepika) [reply scripts/openstack.py]: seems not major, do you want me to add it back with different commit message?

File: docs/detailed_test_config.rst (hunk 3 above)
Comment (dmick) [reply docs/detailed_test_config.rst:240]: extra comma after "fails".  trailing "so" can be removed.

File: docs/detailed_test_config.rst [hunk 5]
```diff
@@ -238,6 +238,11 @@ called after it. This means that before any cleanup happens, you get a
 chance to inspect the system -- both through Teuthology and via extra
 SSH connections -- and the cleanup completes only when you choose so.
 Just exit the interactive Python session to continue the cleanup.
+The interactive-on-error could be enabled with ``--interactive-on-error`` option
+with ``teuthology`` command.
+
+Note: It is advicable to use interactive-on-error mode with ``teuthology``
```
Comment (dmick) [inline docs/detailed_test_config.rst:244]: "advisable", but this whole Note is questionable IMO; should be obvious

File: docs/detailed_test_config.rst [hunk 6]
```diff
@@ -238,6 +238,11 @@ called after it. This means that before any cleanup happens, you get a
 chance to inspect the system -- both through Teuthology and via extra
 SSH connections -- and the cleanup completes only when you choose so.
 Just exit the interactive Python session to continue the cleanup.
+The interactive-on-error could be enabled with ``--interactive-on-error`` option
```
Comment (dmick) [inline docs/detailed_test_config.rst:241]: "You can enable interactive-on-error with the teuthology option --interactive-on-error"

File: scripts/run.py [hunk 7]
```diff
@@ -23,6 +23,10 @@
   --name NAME                    name for this teuthology run
   --suite-path SUITE_PATH        Location of ceph-qa-suite on disk. If not specified,
                                  it will be fetched
+  --interactive-on-error         drop to a python shell on failure, which will keep
+                                 job in halt state, developer can then ssh to
```
Comment (dmick) [inline scripts/run.py:27]: "keep job in halt state" -> halt the job
comma should be semicolon

File: scripts/run.py [hunk 8]
```diff
@@ -23,6 +23,10 @@
...
+                                 target machine and examine the state of cluster.
```
Comment (dmick) [inline scripts/run.py:28]: "examine cluster state"

File: docs/detailed_test_config.rst (hunk 3 above)
Comment (ideepika) [reply docs/detailed_test_config.rst:244]: @dmick updated

File: docs/detailed_test_config.rst (hunk 5 above)
Comment (ideepika) [reply docs/detailed_test_config.rst]: hmm, seems fair, since it's developer centric.

File: scripts/run.py (hunk 7 above)
Comment (ideepika) [reply scripts/run.py]: updated.

File: docs/detailed_test_config.rst [hunk 9]
```diff
@@ -65,6 +61,14 @@ teuthology on it like this::
 
     ./virtualenv/bin/teuthology example.yaml
 
+It is possible to configure installation so that specifying targets and host
+keys could be omitted. Teuthology is run with the ``--lock`` option which locks
```
Comment (dmick) [inline docs/detailed_test_config.rst:65]: could -> can

Comment (ideepika) [reply docs/detailed_test_config.rst]: ack

**Comment by kshtsk** [comment]
@susebot run deploy

**Review by kshtsk — APPROVED** [review]
lgtm

**Comment by kshtsk** [comment]
@dmick are we ready for merge?

**Comment by dmick** [comment]
Yes, it looks good to me.

**Comment by kshtsk** [comment]
Thank you everyone.
